src/domain/event/formSections/timeSection/TimeSection.tsx

```diff
@@ -1,20 +1,58 @@
 import { useField } from 'formik';
+import { IconCalendarPlus } from 'hds-react';
 import React from 'react';
 import { useTranslation } from 'react-i18next';
 
+import Button from '../../../../common/components/button/Button';
+import FormGroup from '../../../../common/components/formGroup/FormGroup';
 import InputRow from '../../../../common/components/inputRow/InputRow';
+import Modal from '../../../../common/components/modal/Modal';
 import Notification from '../../../../common/components/notification/Notification';
+import { INPUT_MAX_WIDTHS } from '../../../../constants';
 import { EVENT_FIELDS } from '../../constants';
 import styles from '../../eventPage.module.scss';
+import { RecurringEventSettings } from '../../types';
+import FieldArrayRow from '../FieldArrayRow';
 import EventTime from './EventTime';
 import EventTimes from './EventTimes';
+import RecurringEvents from './RecurringEvents';
+import RecurringEventsForm from './recurringEventsForm/RecurringEventsForm';
 
 const TypeSection = () => {
   const { t } = useTranslation();
+  const [isModalOpen, setIsmodalOpen] = React.useState(false);
   const [{ value: type }] = useField(EVENT_FIELDS.TYPE);
+  const [
+    { value: recurringEvents },
+    ,
+    { setValue: setRecurringEvents },
+  ] = useField(EVENT_FIELDS.RECURRING_EVENTS);
+
+  const openModal = () => {
+    setIsmodalOpen(true);
+  };
+
+  const closeModal = () => {
+    setIsmodalOpen(false);
+  };
+
+  const addRecurringEventSettings = (values: RecurringEventSettings) => {
+    recurringEvents.push(values);
+    setRecurringEvents(recurringEvents);
```

Comment:
Could this be
`setRecurringEvents([...recurringEvents, values]);`
to avoid mutation

Reply:
Oh yes, changed